app/uk/gov/hmrc/transitmovements/repositories/DeparturesRepository.scala

```diff
@@ -112,6 +128,30 @@ class DeparturesRepositoryImpl @Inject() (
     })
   }
 
+  def getDepartureMessageIds(eoriNumber: EORINumber, departureId: DepartureId): EitherT[Future, MongoError, Option[NonEmptyList[MessageId]]] = {
+
+    val selector = mAnd(mEq("_id", departureId.value), mEq("enrollmentEORINumber", eoriNumber.value))
+
+    val aggregates =
+      Seq(
+        Aggregates.filter(selector),
+        Aggregates.group(null, Accumulators.push("result", "$messages.id")),
```

Comment:
I know it works but think I need an explanation on this .group  to fully understand, please!

Reply:
https://www.mongodb.com/docs/manual/reference/operator/aggregation/group/#std-label-null-example

It's using the aggregation to create a new array that has all the message IDs pushed to it. It basically takes each object, gets the ID field in the messages subdocument, pushes them to a new array named "result", and returns only that. Though there should only be one document at this stage and so there may be a projection may work, there were issues in getting the `replaceRoot` to work with an array, so this was easier.

The below two methods turns the document with a results array into an arrray of documents with one ID each, I believe.

It might be worth opening MongoDB Compass or your favourite Mongo workbench and seeing how it fits together.